arch/arc/core/fast_irq.S

```diff
@@ -101,20 +101,22 @@ firq_nest:
 	 * save original value of _ARC_V2_USER_SP and ilink into
 	 * the stack of interrupted context first, then restore them later
 	 */
-	st ilink, [sp]
-	lr ilink, [_ARC_V2_USER_SP]
-	st ilink, [sp, -4]
+	push ilink
+	pushax ilink, _ARC_V2_USER_SP
+
 	/* sp here is the sp of interrupted context */
 	sr sp, [_ARC_V2_USER_SP]
 
 	/* switch back to banked reg, only ilink can be used */
 	lr ilink, [_ARC_V2_STATUS32]
 	or ilink, ilink, _ARC_V2_STATUS32_RB(1)
 	kflag ilink
+
```

Comment:
Why adding newline here?
If that change doesn't add anything functional to the fix please move it to a separate clean-up patch to preserve clean history.